warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/reindex/ShardReindexMapper.java

```diff
@@ -0,0 +1,837 @@
+     * @param key
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    private void processFI(Context context, Key key) throws IOException, InterruptedException {
+        final byte[] cf = key.getColumnFamilyData().getBackingArray();
+
+        // check if it's the same target field as the last one
+        final int fiBaseLength = FI_START_BYTES.length;
+        final int fiBaseOffset = fiBaseLength + 1;
+
+        // quickly compare the bytes against the last processed bytes to save on parse time if possible
+        if (this.lastFiBytes == null || WritableComparator.compareBytes(cf, fiBaseOffset, cf.length - fiBaseOffset, this.lastFiBytes, fiBaseOffset,
+                        this.lastFiBytes.length - fiBaseOffset) != 0) {
+            // get the field from the cf
+            this.normalizedFieldName = new String(cf, fiBaseLength, cf.length - fiBaseLength);
+            this.lastFiBytes = cf;
+        }
+
+        // parse the dataType from the cq
+        final byte[] cq = key.getColumnQualifierData().getBackingArray();
+        String uid = null;
+        String dataType = null;
+        ByteSequence fieldValue = null;
+        int cqLen = cq.length;
+        int uidNull = -1;
+        for (int i = cqLen - 1; i >= 0; i--) {
+            if (cq[i] == '\u0000') {
+                if (uid == null) {
+                    uid = new String(cq, i + 1, cqLen - i - 1);
+                    uidNull = i;
+                } else if (dataType == null) {
+                    dataType = new String(cq, i + 1, uidNull - i - 1);
+                    fieldValue = key.getColumnQualifierData().subSequence(0, i);
+                    break;
+                }
+            }
+        }
+
+        // get the type from the registry or create it if not already created. There is a cache inside the Type class
+        Configuration config = context.getConfiguration();
+        IngestHelperInterface helper = getIngestHelper(dataType, config);
+
+        if (helper == null) {
+            log.error("cannot find IngestHelperInterface for dataType: " + dataType + " key: " + key);
+            throw new IllegalStateException("cannot find IngestHelperInterface for dataType: " + dataType + " key: " + key);
+        }
+
+        incrementCounter("fi.dataTypes", dataType);
+        incrementCounter("fi.fields", this.normalizedFieldName);
+
+        Text fieldValueText;
+        Text fieldText = null;
+        Text indexCq = null;
+        boolean indexed = false;
+
+        if (key.isDeleted() && !this.propagateDeletes) {
+            incrementCounter("deletes", "skipped");
+            return;
+        } else if (key.isDeleted()) {
+            incrementCounter("deletes", "propagated");
+        }
+
+        // if the field is indexed and index only or events aren't being reprocessed
+        if (helper.isIndexedField(this.normalizedFieldName) && (!this.reprocessEvents || helper.isIndexOnlyField(this.normalizedFieldName))) {
```

Comment:
You might want to move:
```
(!this.reprocessEvents || helper.isIndexOnlyField(this.normalizedFieldName))
```

to a single if statment tha encompasses the ```(helper.isIndexedField(this.normalizedFieldName) ``` and ```(helper.isReverseIndexedField(this.normalizedFieldName)``` checks.